### PR TITLE
docs: Document the testing done in the infra repo

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [x] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

ref https://github.com/adoptium/infrastructure/issues/2502#issuecomment-1766226058

### CODE REVIEW

File: docs/Testing.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- [MacOS 11](https://github.com/adoptium/infrastructure/blob/master/.github/workflows/build_mac.yml)
+- [Windows 2019 and 2022](https://github.com/adoptium/infrastructure/blob/master/.github/workflows/build_wsl.yml)
+
+### Docker Build Images
+
+This workflow job executes playbook changes inside a [CentOS 6](https://github.com/adoptium/infrastructure/blob/master/ansible/docker/Dockerfile.CentOS6) and [Alpine 3](https://github.com/adoptium/infrastructure/blob/master/ansible/docker/Dockerfile.Alpine3) docker container which it builds during the job.
+
+The workflow file can be found [here](https://github.com/adoptium/infrastructure/blob/master/.github/workflows/build.yml).
+
+### QEMU Playbook Check
```
Comment (sxa) [inline docs/Testing.md:47]: This seems to be a duplciate of an earlier section but with different OS versions

File: docs/Testing.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+## Qemu Playbook Check
+
+The QEMU Playbook Check (QPC) job is similar to the VPC job except it is not limited to x86-64 virtual machines.
+
+It is built in our [Jenkins server](https://ci.adoptium.net/job/QEMUPlaybookCheck/). The job itself runs [this script](https://github.com/adoptium/infrastructure/blob/master/ansible/pbTestScripts/qemuPlaybookCheck.sh).
+
+Currently, the supported platorms are:
+- Ubuntu 18.04 on ppc64le, aarch64 and s390x
+- Debian 11 on risc-v
```
Comment (sxa) [inline docs/Testing.md:26]: Suggested change to make it more explicitl, like the other ones (There is a 32-bit RISC-V standard too but that's not what we're using here)
```suggestion
- Debian 11 on riscv64
```

File: docs/Testing.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Testing for Infrastructure Tools
+
+This document will include all of the ways in which testing is done in the infrastructure repository. 
+
+## Vagrant Playbook Check
+
+The Vagrant Playbook Check (VPC) job is used to test changes made to the [windows and linux playbooks](https://github.com/adoptium/infrastructure/tree/master/ansible/playbooks).
```
Comment (sxa) [inline docs/Testing.md:7]:
```suggestion
The Vagrant Playbook Check (VPC) job is used to test changes made to the [Windows and Linux playbooks](https://github.com/adoptium/infrastructure/tree/master/ansible/playbooks).
```

File: docs/Testing.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+These jobs run automatically when a pull request is submitted. The files for these jobs can be found [here](https://github.com/adoptium/infrastructure/tree/master/.github/workflows). 
+
+### Platforms
+
+If a pull request contains changes to the playbooks, this workflow will be triggered automatically. It will execute the playbook changes on the supported platforms:
+- [Solaris 10](https://github.com/adoptium/infrastructure/blob/master/.github/workflows/build_vagrant.yml)
+- [MacOS 11](https://github.com/adoptium/infrastructure/blob/master/.github/workflows/build_mac.yml)
+- [Windows 2019 and 2022](https://github.com/adoptium/infrastructure/blob/master/.github/workflows/build_wsl.yml)
+
+### Docker Build Images
```
Comment (sxa) [inline docs/Testing.md:41]: May be worth noting that these two actions perform a subset of the required build platforms which are pushed from jenkins and this is covered in https://github.com/adoptium/infrastructure/blob/master/FAQ.md#what-about-the-builds-that-use-the-dockerbuild-tag

**Review by sxa — CHANGES_REQUESTED** [review]
Quick set of suggestions that I'm making while on another call (so I may have more later) but this content is generally good!

File: docs/Testing.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+The job is built in our [Jenkins server](https://ci.adoptium.net/job/VagrantPlaybookCheck/). The job runs a user's pull request onto the supported operating systems using Vagrant virtual machines. The vagrant files for the supported operating systems can be found [here](https://github.com/adoptium/infrastructure/tree/master/ansible/vagrant).
+
+Currently, the supported x86-64 systems are:
+- Windows 2012 and 2022
+- Debian 8 and 10
+- Ubuntu 16.04, 18.04, 20.04 and 22.04
+- CentOS 6, 7 and 8
+- Solaris 10
+
```
Comment (steelhead31) [inline docs/Testing.md:17]:
```suggestion
- Fedora 35
```

Add fedora 35 to list of supported systems in VPC

**Review by steelhead31 — CHANGES_REQUESTED** [review]
LGTM, with the suggestions :)

File: docs/Testing.md (hunk 1 above)
Comment (Haroon-Khel) [reply docs/Testing.md:47]: This one is the github workflow https://github.com/adoptium/infrastructure/blob/master/.github/workflows/build_qemu.yml. The earlier one is the jenkins job. I think both should be included in the docs

Comment (sxa) [reply docs/Testing.md:47]: Fair. Perhaps we should have a 'jenkins jobs' section like the Github one to have them categories appropriately?

File: docs/Testing.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+- Fedora 35
+## Qemu Playbook Check
+
+The QEMU Playbook Check (QPC) job is similar to the VPC job except it is not limited to x86-64 virtual machines.
+
+It is built in our [Jenkins server](https://ci.adoptium.net/job/QEMUPlaybookCheck/). The job itself runs [this script](https://github.com/adoptium/infrastructure/blob/master/ansible/pbTestScripts/qemuPlaybookCheck.sh).
+
+Currently, the supported platorms are:
+- Ubuntu 18.04 on ppc64le, aarch64 and s390x
+- Debian 11 on risc64
```
Comment (sxa) [inline docs/Testing.md:26]: My bad - I missed the `v` on my last suggestion
```suggestion
- Debian 11 on riscv64
```

File: docs/Testing.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+The workflow file can be found [here](https://github.com/adoptium/infrastructure/blob/master/.github/workflows/build.yml).
+
+### QEMU Playbook Check
+
+Similar to the [job that runs in our Jenkins server](https://github.com/adoptium/infrastructure/blob/master/docs/Testing.md#qemu-playbook-check) but this job runs in a github workflow. 
+
+The supported platforms are:
+- Ubuntu 18.04 on aarch64, ppc64le and s390x
+- Ubuntu 20.04 on risc-v
```
Comment (sxa) [inline docs/Testing.md:53]:
```suggestion
- Ubuntu 20.04 on riscv64
```

**Review by sxa — APPROVED** [review]
I'll let you decide whether to implement the changes or not, but overall I'll give an approval so you don't have to wait for me again :-)

**Review by steelhead31 — APPROVED** [review]
LGTM

**Pull request merged**